omegaconf/dictconfig.py

```diff
@@ -42,7 +42,7 @@ def __setitem__(self, key, value):
             )
 
         input_config_or_node = isinstance(value, (BaseNode, Config))
-        if key in self:
+        if self.get_node(key):
```

Comment:
Does this resolve the `Node` or resolve node's value? If the latter then it's not what you want

Reply:
the node itself.

Reply:
The node. iirc the problem is that if the value is ??? this returns false now and it was breaking the semantics I was going for with the new logic.